### ISSUE TITLE
VSCode and folders in repository from GitHub

### ISSUE THREAD

**Issue by espenj90** [microsoft/vscode-java-debug#785]
First of all I'm not an expert, but I will try to explain my issue as well as I can. I'm not sure if it's a bug, but I still need help to figure it out.
I was working with a GitHub-repository with standalone .java-files (no Java-project, Maven etc.). I'm a student, and so far we are programming Java by making different classes in different .java-files, that works together by being in the same folder. Then I changed the name of the repository, and added some folders within the repository and moved my .java-files into one of those folders. Then I tried to run the my program, by hitting "run" on in the file that contain my "static void main-method". It then returned a "fileNotFoundException", because the program I coded was trying to read from a .txt-file inside the folder. The program had worked before I renamed the repository and moved the files into the folder. After some googling, I found out that the issue was that I moved the .txt file, and that this could be fixed by deleting the directory "Library/Application Support/Code/User/workspaceStorage". So I did, and it didn't fix the problem, and created another problem, which is now the problem that I need help with. When I open my GitHub-repository (which I have cloned to my Mac) in VSCode it opens fine, and I can open the folders inside the repository in the file-explorer in VSCode. But when I open one of the .java-files within one of those folders, it seems like VSCode sets this folder to a main folder or something, because when I go into one of those other folders inside the repository, and open one of the .java files there, then VSCode says that "_filename_.java is not on the classpath of project _nameOfProject_, only syntax errors are reported". But I have not created a project or anything, I just have .java-files inside the same folder, and they work together that way. So how do I fix this issue, because I didn't have a problem with this before, and it reported all sort of errors, not only syntax error

##### Environment
- Operating System: MacOS 10.15.4
- JDK version: 1.8.0_231
- Visual Studio Code version: 1.43.2
- Java extension version: 0.8.1
- Java Debugger extension version: 0.25.1

##### Steps To Reproduce
1. Open my GitHub-repository that has been cloned to my computer.
2. Open a .java-file within a subfolder in the repository.
3. Open a .java-file inside another of the subfolders in the repository.
4. Try to run a file with static void main-method in it.

[attach a sample project reproducing the error]
[attach logs]
[I try to add a logg here, hope this is the correct one](https://drive.google.com/open?id=15ypWpqqXLf1J5B73rBDe0HoZwVV1Qnfu)

**Comment by testforstephen** [comment]
Go to File Explorer, try the context menu "**Add Folder to Java Source Path**" on the folder with "not on the classpath" warning. This action will mark the unrecognized folder as a source folder.

**Comment by espenj90** [comment]
Thank you for the reply, but I have already tried this, and this workaround creates another problem. If I have two files called Test.java (as an example) which both contains a class named Test, in two separate folders, and I add both folders to Java Source Path, then I get an error which says that "The type Test is already defined". 
Any other ideas of what I could try?

**Comment by testforstephen** [comment]
I don't think this is the correct approach to define two classes with the same name in the same namespace. This is not allowed by Java language. You must put them in different packages.

See a sample structure below, then you just need mark the `src` folder to Java source path.
```
- src
  - packageA
     - Test.java  // declare "package packageA;" in the header
  - packageB
     - Test.java // declare "package packageB;" in the header
```

**Comment by espenj90** [comment]
Yes, it might be that I'm doing something wrong, I don't really doubt that. I can try to explain myself a little better, so perhaps it's easier to figure out if it's possible to find a fix for me, or not.
So my GitHub repository is called "IN1010", and inside that repository I have folders named Oblig4, Oblig5, Oblig6. In VSCode I just open my repository (IN1010), and then in the file explorer I can se all the folders (oblig4, oblig5, oblig6). Inside those folders I have several .java-files, which contains a lot of classes and together form a program inside that one folder. So you could say that I have one complete program inside each folder, and the programs don't have anything to do with the programs in the other folders. Every .java-file start like this: 
```
class Classname {
}
```
So I don't have any project, package, src or anything. Does that help in understanding my problem, and to understand if there is a fix that I could try?
Because then if I have a Example.java-file (with class Example{} inside it) in both oblig4 and oblig5-folder, I get the error like I told you about ("The type Example is already defined"). This is even if they are in separate folders, and have nothing to do with each other.

**Comment by testforstephen** [comment]
**Impossible**.  

From the perspective of the build tool, what most concerned is about **classpath**. project, package, src are all used to help define the classpath. You didn't declare any packages, that doesn't mean there are no packages, but all classes are placed at the default package space. Name conflicting is not allowed at the same package. 

When you opened multiple folders in the same VS Code window, then the classes in these folders are visible to each other. Either use different packages to isolate classes of the same name, or just open a subfolder at a time.

**Comment by testforstephen** [comment]
You may take a look at multi-root workspace. https://code.visualstudio.com/docs/editor/multi-root-workspaces. Root will be considered as a kind of isolation mechanism. Classes with same name in different root are allowed.

**Comment by espenj90** [comment]
Ok. Thank you for the suggestion, I will try that.
